[android] RN 0.77 support

## Issue: React Native 0.77 Kotlin Compiler v2 Compatibility Issues with Didomi SDK

### Description
After upgrading to React Native 0.77, which introduced Kotlin compiler v2 for Android, the Didomi SDK is failing to compile due to null safety violations in Kotlin code.

### Environment
- React Native: 0.77
- Android Gradle Plugin: 8.7.2
- Gradle: 8.10.2
- Didomi React Native SDK version: 2.10.0

### Error Message

```
Android gradle plugin: 8.7.2
Gradle: 8.10.2

> Task :didomi_react-native:compileDebugKotlin FAILED
e: file:///Users/valentin/workspace/app/node_modules/@didomi/react-native/android/src/main/java/io/didomi/reactnative/DidomiModule.kt:271:21 Argument type mismatch: actual type is 'kotlin.String?', but 'kotlin.String' was expected.
e: file:///Users/valentin/workspace/app/node_modules/@didomi/react-native/android/src/main/java/io/didomi/reactnative/DidomiModule.kt:1182:51 Argument type mismatch: actual type is 'kotlin.String?', but 'kotlin.String' was expected.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':didomi_react-native:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```

### Root Cause
The error occurs because Kotlin compiler v2 is more strict about null safety. The SDK is attempting to pass potentially nullable strings (`String?`) to methods that expect non-null strings (`String`).

### Steps to Reproduce
1. Use React Native 0.77 or higher
2. Install @didomi/react-native SDK
3. Attempt to build Android app

### Expected Behavior
The SDK should compile successfully with Kotlin compiler v2.

### Suggested Fix
The SDK needs to be updated to handle null safety properly in the affected methods at:
- DidomiModule.kt:271
- DidomiModule.kt:1182

Would you like me to create a pull request to fix these null safety issues?

--- a/android/src/main/java/io/didomi/reactnative/DidomiModule.kt
+++ b/android/src/main/java/io/didomi/reactnative/DidomiModule.kt
@@ -268,7 +268,7 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
         val set = mutableSetOf<String>()
 
         for (i in 0 until size()) {
-            set.add(getString(i))
+            getString(i)?.let { set.add(it) }
         }
 
         return set
@@ -1179,7 +1179,7 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
 
     // Required to transform from array to variadic.
     private fun readableArrayToStringArray(readableArray: ReadableArray): Array<String> {
-        return Array(readableArray.size()) { i -> readableArray.getString(i) }
+        return Array(readableArray.size()) { i -> readableArray.getString(i) ?: "" }
     }
 
     private fun buildDidomiInitializeParameters(jsonParameters: String): DidomiInitializeParameters {

--- a/android/src/main/java/io/didomi/reactnative/DidomiModule.kt
+++ b/android/src/main/java/io/didomi/reactnative/DidomiModule.kt
@@ -268,7 +268,7 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
         val set = mutableSetOf<String>()
 
         for (i in 0 until size()) {
-            getString(i)?.let { set.add(it) }
+            getString(i)?.also { set.add(it) }
         }
 
         return set
@@ -1178,9 +1178,7 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
     }
 
     // Required to transform from array to variadic.
-    private fun readableArrayToStringArray(readableArray: ReadableArray): Array<String> {
-        return Array(readableArray.size()) { i -> readableArray.getString(i) ?: "" }
-    }
+    private fun readableArrayToStringArray(readableArray: ReadableArray) = readableArray.toSet().toTypedArray()
 
     private fun buildDidomiInitializeParameters(jsonParameters: String): DidomiInitializeParameters {
         val jsonObject = JSONObject(jsonParameters)

